test_tracetools_launch/test/test_tracetools_launch/test_trace_action.py

```diff
@@ -80,8 +80,11 @@ def _assert_launch_frontend_no_errors(self, file: TextIO) -> Tuple[Trace, Launch
         ls = LaunchService()
         ls.include_launch_description(ld)
         self.assertEqual(0, ls.run(), 'expected no errors')
-        trace_action = ld.describe_sub_entities()[0]
-        assert isinstance(trace_action, Trace), f'expected Trace action, got: {trace_action}'
+        trace_action = next(
+            (action for action in ld.entities if isinstance(action, Trace)),
+            None
+        )
+        assert trace_action is not None, f'expected Trace action, got: {trace_action}'
```

Comment:
Nice! Just a minor suggestion, since the assertion condition changed a bit: we're now asserting that we've found the action amongst the launch entitites

```suggestion
        trace_action = next(
            (action for action in ld.entities if isinstance(action, Trace)),
            None
        )
        assert trace_action is not None, 'did not find Trace action '
```

Reply:
done!